lib/streamlit/elements/widgets/camera_input.py

```diff
@@ -33,18 +33,18 @@
     WidgetKwargs,
     register_widget,
 )
-from streamlit.runtime.uploaded_file_manager import UploadedFile, UploadedFileRec
+from streamlit.runtime.uploaded_file_manager import DeletedFile, UploadedFile
 from streamlit.type_util import Key, LabelVisibility, maybe_raise_label_warnings, to_key
 
 if TYPE_CHECKING:
     from streamlit.delta_generator import DeltaGenerator
 
-SomeUploadedSnapshotFile = Optional[UploadedFile]
+SomeUploadedSnapshotFile = Union[UploadedFile, DeletedFile, None]
 
 
-def _get_file_recs_for_camera_input_widget(
-    widget_id: str, widget_value: Optional[FileUploaderStateProto]
-) -> List[UploadedFileRec]:
+def _get_upload_files(
```

Comment:
Can we factor this out into a helper function shared by `camera_input` and `file_uploader`? This function seems to be identical for both of the widgets

Reply:
Yes, it difficult to find a new home for `def _get_upload_files` (because it is used only in file_uploader and camera_input, so `utils.py` is probably not the best place, so I just now imported `_get_upload_files` from file_uploader for camera_input,  I think it is good idea, and denote the fact that camra_input essentially a derivative form file_uploader